core/consensus/src/execution/transaction_store.rs

```diff
@@ -93,11 +93,14 @@ impl<T: BroadcastEventInterface<PubSubMsg>> TransactionStore<T> {
         self.store_attestation_internal(self.next_pointer(), digest, node_index, Some(event));
     }
 
-    // When the epoch changes, the parcels from the current epochs become
-    // the parcels from the previous epoch.
-    // The parcels from the previous epoch are garbage collected, and the parcels from
-    // the next epoch become the parcels from the current epoch (after validating them).
-    pub fn change_epoch(&mut self, committee: &[NodeIndex]) {
+    /// Change the committee.
+    ///
+    /// When the committee is changed for across epoch eras and epochs, the parcels from the current
+    /// become the parcels from the previous.
+    ///
+    /// The parcels from the previous are garbage collected, and the parcels from the next becom
+    /// ethe parcels from the current after validating them.
+    pub fn change_committee(&mut self, committee: &[NodeIndex]) {
```

Comment:
I left this comment 2 days ago, it said "PENDING" next to it. I thought that meant that the review is pending, but turns out the comment wasn't even posted lol.

Reply:
That reasoning seems sound to me, but I'm still wrapping my head around it and wondering if I can write a test to mimic the scenario.

Is https://github.com/fleek-network/lightning/pull/65/commits/3b0c7acb8d0efb9a28b3b9bea6960a56ca534cd3 what you had in mind for the change? Tests are still green with it

Reply:
Yeah this is exactly what I had in mind. We just have to make sure that this won't cause a node to wrongfully reject a parcel or something like that. I don't think this can happen, but I will also go over it again.

Reply:
I went over it again, and I think we do need to make some changes to the transaction store.

- lets say we are in epoch `n` and era `0`
- a node is slashed and we move to era `1`
- parcels `p1` and `p2` are from era `0` and parcel `p3` is from era `1` with the new committee
- `p2` contains the transaction that triggers the era change
- node `A` that isn't on the committee receives the parcels out of order: `p1`, `p3`, `p2`
- when node `A` receives `p1`, it checks the originator, which is from the committee in era `0`. since node `A` is still on era `0` it correctly accepts the parcel
- node `A` receives `p3`, which originates from a validator `v` in era 1, but node `A` is still on era `0`
- there three possibilities:
1. validator `v` was both on the committee in era `0` and era `1`  
   - in this case node `A` will correctly accept parcel `p3` since `v` is part of the committee in era 0 (which is node `A`'s current committee)
   - we don't have to make any changes for this case
2. validator `v` was not part of the committee in era `0`, but is on the committee in era 1
   - then node A will reject `p3`, even though it is a valid parcel
   - however, as I understand it, this cannot happen, because for any two eras `e1` and `e2` in some epoch `n`: if `e1` < `e2`, then the committee in `e2` is a subset of the committee in `e1`, so there will never be a node in the committee of epoch `e2` that wasn't on the committee in epoch `e1`.
   - we don't have to make any changes for this case
3. validator `v` was on the committee in era `0`, but not on the committee in era `1` (`v` was slashed and kicked)
   - in this case `p3` is not a valid parcel
   - however, node A will accept `p3` (since `A` is still in era `0` and v was on the committee in era `0`)
   - we have to make changes for this case
   - in the transaction store instead of having `previous epoch | current epoch | next epoch` we can change it to `previous epoch | current epoch | current epoch (pending) | next epoch`
   - on every era change, we validate the parcels and attestations in `current epoch (pending)` the way we do it with the parcels and attestations in `next epoch`
   - the valid parcels and attestations are moved to `current epoch`
   
  Please check my logic here. 
  
  If my logic is sound here, do you want me to push this change? I'm fine either way

Reply:
That logic makes sense to me. Yep if you're able to push the change that'd be great, appreciate it 🙏

Reply:
Once I started implementing this, I realized that this won't work in case there aren't any era changes in an epoch. Because then the parcels/attestations from `current epoch (pending)` are never moved to `current epoch`.
However, I also realized that we don't necessary need this, because in case 3 from my comment above, `p3` will be stored in the transaction store even though it is an invalid parcel, but it won't receive `2f+1` attestations, so it won't be executed.

What we probably should do is on every era change, go over the parcels/attestations from the current epoch and throw out parcels/attestations like `p3` that originate from nodes that aren't on the committee anymore.
But I can push this up once you merge this PR, since this is optimization/garbage collection and not critical for functionality.